server/WAPI.rb

```diff
@@ -252,6 +272,31 @@ def get_request(request)
         wrapped_response = do_request(request)
       end
     end
+
+    # If it didn't work just return that information.
+    logger.debug "#{self.class.to_s}:#{__method__}:#{__LINE__}: wrapped_response: meta_status: #{wrapped_response.meta_status}"
+    if wrapped_response.meta_status != WAPI::SUCCESS
+      return wrapped_response
+    end
+
+    # Ran a query successfully.  See if got partial data and need to keep going.
```

Comment:
Since @zqian commented about the documentation, I feel I need to ask about some of the comments.  This one, for example says that there's a check being done.  Actually, it's just logging something.  Is this comment old or just inaccurate?  I've heard other developers say comments should be avoided because they cause confusion when they're inaccurate.  Maybe they start out correct, but after the code has changed and nobody remembered to update the comment correspondingly, then they can cause confusion.